### ISSUE TITLE
Middleware generator problems

### ISSUE THREAD

**Issue by combsco** [keithwhor/nodal#126]
### Not extending Nodal.Middleware in generated file

```nodal g:middleware IsAuthed```

```
module.exports = (function() {

  "use strict";

  const Nodal = require('nodal');

  class IsAuthedMiddleware {

    exec(controller, data, callback) {

      callback(null, data);
      return false;

    }

  }

  return IsAuthedMiddleware;

})();
```

### Supplying ```exec``` with params ```contoller, data, callback```
The nodal docs state this is the correct format but this gives back the error that callback is not a function. What does works is ``` exec(controller, callback)``` which you can find in the cors_middleware.js file.

Also it doesn't auto add the new generated middleware to app.js but that might be intentional.

**Comment by intabulas** [comment]
FYI I fixed this yesterday in #118, which came **after** the 0.5.7 release.. @keithwhor this can be closed

**Comment by keithwhor** [comment]
Ah shit. So `Middleware`, `Renderware` and `Initializers` *aren't supposed to be extended*. They're sitting there as example templates. This is to make it really easy to write separate npm modules without `peerDependencies`, which are an effing nightmare. :)

@intabulas Will have to fix that. Sorry, missed that change or would've caught it on the PR.

**Comment by intabulas** [comment]
@keithwhor okay that makes alot of sense.. I will revert them in a new PR before 0.6

**Comment by intabulas** [comment]
@keithwhor I guess we need to remove the extends from the gzip and cors in src/ as well

**Comment by intabulas** [comment]
@keithwhor we can close this #118 fixed it and #128 removed the extends
